contributing-docs/testing/unit_tests.rst

```diff
@@ -0,0 +1,1160 @@
+
+We have also tests that run "all" tests (so they do not look at the folder, but at the ``pytest`` markers
+the tests are marked with to run with some filters applied.
+
+* ``All-Postgres`` - tests that require Postgres database. They are only run when backend is Postgres (``backend("postgres")`` marker)
+* ``All-MySQL`` - tests that require MySQL database. They are only run when backend is MySQL (``backend("mysql")`` marker)
+* ``All-Quarantined`` - tests that are flaky and need to be fixed (``quarantined`` marker)
+* ``All`` - all tests are run (this is the default)
+
+
+We also have ``Integration`` tests that are running Integration tests with external software that is run
+via ``--integration`` flag in ``breeze`` environment - via ``breeze testing integration-tests``.
+
+* ``Integration`` - tests that require external integration images running in docker-compose
+
+This is done for three reasons:
+
+1. in order to selectively run only subset of the test types for some PRs
+2. in order to allow efficient parallel test execution of the tests on Self-Hosted runners
+
+For case 2. We can utilise memory and CPUs available on both CI and local development machines to run
+test in parallel, but we cannot use pytest xdist plugin for that - we need to split the tests into test
+types and run each test type with their own instance of database and separate container where the tests
+in each type are run with exclusive access to their database and each test within test type runs sequentially.
+By the nature of those tests - they rely on shared databases - and they update/reset/cleanup data in the
+databases while they are executing.
+
+
+DB and non-DB tests
+-------------------
+
+There are two kinds of unit tests in Airflow - DB and non-DB tests.
+
+Some of the tests of Airflow (around 7000 of them on October 2023)
+require a database to connect to in order to run. Those tests store and read data from Airflow DB using
+Airflow's core code and it's crucial to run the tests against all real databases that Airflow supports in order
+to check if the SQLAlchemy queries are correct and if the database schema is correct.
+
+Those tests should be marked with ``@pytest.mark.db`` decorator on one of the levels:
+
+* test method can be marked with ``@pytest.mark.db`` decorator
+* test class can be marked with ``@pytest.mark.db`` decorator
+* test module can be marked with ``pytestmark = pytest.mark.db`` at the top level of the module
+
+Airflow's CI runs different test kinds separately.
+
+For the DB tests, they are run against the multiple databases Airflow support, multiple versions of those
+and multiple Python versions it supports. In order to save time for testing not all combinations are
+tested but enough various combinations are tested to detect potential problems.
+
+As of October 2023, Airflow has ~9000 Non-DB tests and around 7000 DB tests.
+
+Airflow non-DB tests
+....................
+
```

Comment:
Cound of DB tests is mentioned above already, would not repeat this here. Also would move it _into_ the chapter.
```suggestion
Airflow non-DB tests
....................

As of October 2023, Airflow has ~9000 Non-DB tests.

```

Reply:
Removed the numbers altogether. It adds no value.